lib/abr/simple_abr_manager.js

```diff
@@ -333,7 +380,7 @@ shaka.abr.SimpleAbrManager = class {
 
         return shaka.util.StreamUtils.meetsRestrictions(
             variant, restrictions,
-            /* maxHwRes= */ {width: Infinity, height: Infinity});
+            /* maxHwRes= */ {width: maxWidth, height: maxHeight});
```

Comment:
`shaka.util.StreamUtils.meetsRestrictions` already filters on `restrictions.maxWidth` and `restrictions.maxHeight`, why add new args?

Reply:
This is the part where you had the most doubts, and I'm glad you mentioned it :)

The problem is the mutation of objects, because the restrictions come from the configuration (which can be dynamic) and apart from that the size of the video can be dynamic. So there were two possible solutions:
1) the one that is implemented
2) Create a new restrictions object and calculate the minimum of restrictions.maxWidth/maxHeight and this.mediaElement_.clientWidth/clientHeight

If your proposal is to implement the second option, I will gladly do it

@littlespex I await your comments

Reply:
Looks good to me, since you're using existing arguments to StreamUtils from AbrManager, and we were not using them in this context previously.

Reply:
> The problem is the mutation of objects,
Good point. Option 1 is the way to go.